pkg/state/interface.go

```diff
@@ -0,0 +1,11 @@
+package state
+
+import "github.com/redhat-developer/odo/pkg/api"
+
+type Client interface {
+	// SetForwardedPorts sets the forwarded ports in the state file and saves it to the file, updating the metadata
+	SetForwardedPorts(fwPorts []api.ForwardedPort) error
```

Comment:
Thinking out loud - why do we want business layer to get involved with storing the state? Isn't this something that odo "CLI" is interested in doing? Maybe a different implementation using odo as library would not want to store the state.

Reply:
In any case, we want to create an abstraction for the state, hence this package. 
For me, it is a business concern, as it is a way to be able to give to the CLI the state of the Dev session at any time. Any other CLI would be interested in getting this state.